src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +316,1319 @@ const ActorInit En_Egol_InitVars = {
     (ActorFunc)EnEgol_Draw,
 };
 
-#endif
+void EnEgol_SetAnimation(EnEgol* this, s32 animation) {
+    static AnimationHeader* sAnimations[EYEGORE_ANIM_MAX] = {
+        &gEyegoreStandAnim,    &gEyegoreWalkAnim,      &gEyegoreSlamAnim,
+        &gEyegoreSlamWaitAnim, &gEyegoreSlamEndAnim,   &gEyegoreDamagedAnim,
+        &gEyegoreDeathAnim,    &gEyegoreLaserAnim,     &gEyegoreUnusedLaserEndAnim,
+        &gEyegoreStunnedAnim,  &gEyegoreStunEndAnim,   &gEyegoreRetreatAnim,
+        &gEyegoreSitAnim,      &gEyegoreLeftPunchAnim, &gEyegoreRightPunchAnim,
+    };
+    static u8 sAnimModes[EYEGORE_ANIM_MAX] = {
+        ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE,
+        ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE,
+        ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE,
+    };
+    this->animation = animation;
+    this->animEndFrame = Animation_GetLastFrame(sAnimations[this->animation]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animation], 1.0f, 0.0f, this->animEndFrame,
+                     sAnimModes[this->animation], 0.0f);
+}
+
+void EnEgol_FootstepEffects(EnEgol* this, PlayState* play, f32 leftFootFrame, f32 rightFootFrame) {
+    if (Animation_OnFrame(&this->skelAnime, leftFootFrame) || Animation_OnFrame(&this->skelAnime, rightFootFrame)) {
+        Vec3f spawnPos;
+        Player* player = GET_PLAYER(play);
+        s32 quakeSize;
+        s32 pad;
+
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_EYEGOLE_WALK);
+        quakeSize = 4 - (s32)(fabsf(player->actor.world.pos.y - this->actor.world.pos.y) * 0.02f);
+        if (quakeSize > 4) {
+            quakeSize = 4;
+        } else if (quakeSize < 1) {
+            quakeSize = 1;
+        }
+
+        if (player->stateFlags3 != PLAYER_STATE3_1000000) {
+            func_800BC848(&this->actor, play, quakeSize, 2);
+        }
+        if (Animation_OnFrame(&this->skelAnime, leftFootFrame)) {
+            Math_Vec3f_Copy(&spawnPos, &this->leftFootPos);
+            spawnPos.y = this->actor.floorHeight;
+            Actor_SpawnFloorDustRing(play, &this->actor, &spawnPos, 0.0f, 10, 6.0f, 50, 30, true);
+        } else {
+            Math_Vec3f_Copy(&spawnPos, &this->rightFootPos);
+            spawnPos.y = this->actor.floorHeight;
+            Actor_SpawnFloorDustRing(play, &this->actor, &spawnPos, 0.0f, 10, 6.0f, 50, 30, true);
+        }
+    }
+}
+
+void EnEgol_DestroyBlocks(EnEgol* this, PlayState* play, Vec3f pos1, Vec3f pos2) {
+    Actor* egBlock = play->actorCtx.actorLists[ACTORCAT_PROP].first;
```

Comment:
What do you think about calling this `prop` instead of `egBlock`? The logic being that there are plenty of things in here that *aren't EgBlocks that it will be iterating through.